### PR TITLE
Suggested changes to RAP pages on the analysts guide

### PR DESCRIPTION
**Pull request by jen-machin**
## Overview of changes
I added some extra details and fixed some typos (sorry) on all 4 pages. 

On the **RAP for Statistics page** I have quite a few suggested changes but I didn't want to make these without running them by someone first. They are:
- I think this page should come first in the list of RAP pages as it's a good overview
- I don't know whether it's maybe worth adding a table of contents linking to each of the RAP principles, but this might be too long/unwieldy given how many of them there are. Might be useful for people looking for a specific section though. 
- The diagram under "Where do we need to focus" is too small and difficult to read, and you can't expand it/zoom in. The same is true for the diagrams under "Version controlled final code scripts". 
- "How to get started" only refers to publications, do we need more generic info for non-publication related RAP?
- Some of the code snippets on the page show up ok but are difficult to differentiate from the other text despite the change in font, and other code snippets (like the ones under the "Tidying and processing data in R", "Recyclable code for future use", and "Build capability with in the team" sections) are pretty much impossible to read because of the font colours against the dark background. Could we maybe put code snippets inside boxes with a different colour background or something? 
- Under "Version controlled final code scripts", do we need to more clearly differentiate between the use case for DevOps (private) and GitHub (open source)? It's implied but not sure how clear it'd be to people that are new to using both. 

I maybe should have added these as comments on the actual bits of Markdown text. I can go back and do that if it's better 😄 

Also I asked to merge this to main because there's no development branch that I can see - sorry!

### CODE REVIEW

File: RAP/rap-faq.qmd [hunk 1]
```diff
@@ -165,11 +165,11 @@ Questions on how to implement RAP.
...
 
-We're not suggesting all code lives in a single script. These steps in the RAP guidance are to encourage the repository being structured so that one clicks of a button can run the entire process from the source data through to the final output in order.
+We're not suggesting all code lives in a single script. These steps in the RAP guidance are to encourage the repository being structured so that one click of a button can run the entire process from the source data through to the final output in order.
 
-This means that you should have one 'centralized' script that details all of the different scripts in your repository. This single run script then provides a single reference point for where all of your code and scripts live, and in what order they're executed, which is fantastic piece documentation to have!
+This means that you should have one 'centralized' script that details all of the different scripts in your repository. This single run script then provides a single reference point for where all of your code and scripts live, and in what order they're executed, which is a fantastic piece documentation to have!
```
Comment (cjrace) [inline RAP/rap-faq.qmd:172]: should this be 'a fantastic piece of documentation?

File: RAP/rap-faq.qmd [hunk 2]
```diff
@@ -213,7 +213,9 @@ Questions learning more about RAP and developing skills.
 
 Absolutely. Currently the best way to do this is to ask other teams to share code directly with you. If we can all make more progress with using Git, then this will make it much easier to share repositories and have code open by default for interested analysts to browse.
 
-We are also starting up a knowledge share series, so keep an eye out for that and make sure to contribute yourselves. We'd also encourage analysts to make more use of teams to post questions and ask about other people in our area doing similar types of analysis, no question is too big or too small!
+We run a regular RAP Knowledge Share series in which we share good practice and examples of work. We record each session so that you can refer back to them at a later date if you want to. Previous recordings can be found on [Sharepoint](https://educationgovuk.sharepoint.com/:f:/r/sites/lvewp00086/SSU%20%20open%20sharing/Reproducible%20Analytical%20Pipeline%20(RAP)%20resources/Knowledge-share%20series). Upcoming sessions are advertised on the DfE analysts network mailing list.
```
Comment (cjrace) [inline RAP/rap-faq.qmd:216]: Could the link text be something like 'RAP knowledge share sharepoint folder' or something? Just thinking out of context sharepoint might not make sense / or easily clash if we put other links to sharepoint also on this page

File: RAP/rap-statistics.qmd [hunk 3]
```diff
@@ -142,7 +142,7 @@ The Statistics Development Team invites teams to take part in our partnership pr
 
 ---
 
-The first place to start for your teams RAP is to store the raw data you use to create underlying data in a Microsoft SQL Server database. This is similar to a sharepoint area or a shared folder, but it's a dedicated data storage area which allows multiple users to use the same file at once, and for you to run code against the data in one place.
+The first place to start for your teams RAP is to store the raw data you use to create underlying data in a Microsoft SQL Server database. This is similar to a Sharepoint area or a shared folder, but it's a dedicated data storage area which allows multiple users to use the same file at once, and for you to run code against the data in one place.
```
Comment (cjrace) [inline RAP/rap-statistics.qmd:145]: Never occurred to me this should be capitalised - great spot!

**Review by cjrace — CHANGES_REQUESTED** [review]
A few minor suggestions though generally this is excellent! 

In general, very happy for you to just make the changes to the stats RAP page.

The diagram on where we need to focus is probably long out of date now, so happy for you to drop it and we can always ditch that part or update to a new place where we want to focus.

For the code snippets I think we can fix those with a custom css script, Rich and Tess had brief goes without quick success but sure with a bit of time we could add that in and overwrite the defaults we have now to improve contrast

File: RAP/rap-expectations.qmd [hunk 4]
```diff
@@ -10,15 +10,13 @@ title: "RAP expectations"
...
+As government analysts working with statistics, we are required to ensure that our analysis is reproducible, transparent, and robust, using coding and code management best practices (source [GSG competency framework](https://analysisfunction.civilservice.gov.uk/policy-store/competency-framework-for-the-government-statistician-group-gsg/)). Reproducible Analytical Pipelines (RAP) are a cross-government requirement to help analysts adopt best practices.
 
-We expect any analyst working in statistics production to know and be able to implement RAP principles, using the recommended tools to meet at least good and great practice. 
+We expect any analyst working in statistics production to know and be able to implement RAP principles, using the recommended tools to meet at least good and great practice. You can see examples of good, great, and best practice on our [RAP in statistics guidance page](../RAP/rap-statistics.html).
```
[the remark this answers is not in the document]
Comment (jen-machin) [reply RAP/rap-expectations.qmd:15]: I might just drop out the "in statistics production" bit and then it's more applicable to everyone

**Comment by jen-machin** [comment]
> A few minor suggestions though generally this is excellent!
> 
> In general, very happy for you to just make the changes to the stats RAP page.
> 
> The diagram on where we need to focus is probably long out of date now, so happy for you to drop it and we can always ditch that part or update to a new place where we want to focus.
> 
> For the code snippets I think we can fix those with a custom css script, Rich and Tess had brief goes without quick success but sure with a bit of time we could add that in and overwrite the defaults we have now to improve contrast.
> 
> For the ordering, I'm happy for you to change that about I think it makes sense.
> 
> Not sure on the table, as I guess the table of contents on the right should give that? Though we can certainly have a look at it and if there's a way we can make it easier to navigate.

I think I've made all of these changes now 😄 I chose a theme for the code snippets (sadly not Dracula) but happy to change this to a different one if people have a preference. There's info on the different themes [here](https://quarto.org/docs/output-formats/html-code.html#highlighting). Also some of the code blocks were specified as R code and the others weren't - the theming only works on code blocks that are specifically identified as a certain language so I just modified all the blocks to be R blocks. I don't think this has caused any catastrophes but maybe worth checking.

**Review by cjrace — DISMISSED** [review]
Happy to approve - we should take note of this anywhere else we have code chunks to fix the highlighting.

Could you add something to the contributing file about this @jen-machin, just a short sentence or two on how to add in code chunks so the highlighting works for both light and dark matching the theme you've added? Can either be on this branch / PR or in a follow up one

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,25 +1,47 @@
+---
+editor: 
+  markdown: 
+    wrap: 72
+---
```
Comment (cjrace) [inline CONTRIBUTING.md:5]: Not sure this has worked? 
image

I previewed it by switching to the branch then selecting the file, leading me to https://github.com/dfe-analytical-services/analysts-guide/blob/rap-page-additions/CONTRIBUTING.md

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,25 +1,47 @@
...
+
 # Contributing to the Analyst's Guide
 
-Thank you for investing your time in contributing to the analyst's guide! All contributions are very welcome - we want this to be a useful resource for the whole analytical community in DfE :sparkles:.
+Thank you for investing your time in contributing to the analyst's
+guide! All contributions are very welcome - we want this to be a useful
+resource for the whole analytical community in DfE :sparkles:.
```
Comment (cjrace) [inline CONTRIBUTING.md:11]: Did you tidy these line lengths manually or was there a package you used to do it?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -30,52 +52,96 @@ For more information specific to how a quarto site works, and for examples of wh
 - building locally before raising a pull request
 
 -->
```
Comment (cjrace) [inline CONTRIBUTING.md:54]: This is now showing the comment in the preview of the file, was this intentional? 

image

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -30,52 +52,96 @@ For more information specific to how a quarto site works, and for examples of wh
...
+2.  Open the repository in your editor of choice, e.g. R Studio.
 
-3. Create a working branch and start with your changes!
+3.  Create a working branch and start with your changes!
 
-4. Commit and push the changes to your working branch.
+4.  Commit and push the changes to your working branch.
 
-### Pull Request
+### Pull Request {#pull-request}
```
Comment (cjrace) [inline CONTRIBUTING.md:122]: Was this intentionally added?

image

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -30,52 +52,96 @@ For more information specific to how a quarto site works, and for examples of wh
...
-
-If you need any assistance at all, please contact [statistics.development@education.gov.uk](mailto:statistics.development@education.gov.uk).
\ No newline at end of file
+When you're finished with the changes, create a pull request, also known
+as a PR. - Don't forget to [link PR to
+issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
+if you are solving one.
+
+Once you submit your PR, a repository admin will review your proposal.
+We may ask questions or request additional information. - We may ask for
```
Comment (cjrace) [inline CONTRIBUTING.md:132]: Think the bullet points have been lost here? 

Preview of this:
image

What it looks like on main (https://github.com/dfe-analytical-services/analysts-guide/blob/main/CONTRIBUTING.md)

image

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -30,52 +52,96 @@ For more information specific to how a quarto site works, and for examples of wh
...
 
-Key things to remember when making or proposing changes:
-- Where guidance already exists elsewhere it should be linked to rather than duplicated
-- When adding images, save them in the /images folder
-- When adding files for download, save them in the /resources folder
-- Don't commit rendered .html files for pages
-- If your changes remove or edit a pre-existing anchor link, consider how that will be redirected for users who may have bookmarked it
+Key things to remember when making or proposing changes: - Where
+guidance already exists elsewhere it should be linked to rather than
+duplicated - When adding images, save them in the /images folder - When
```
Comment (cjrace) [inline CONTRIBUTING.md:99]: Think we've lost the bullet point formatting, was this intentional? ([your branch](https://github.com/dfe-analytical-services/analysts-guide/blob/rap-page-additions/CONTRIBUTING.md) is on the left and the [main branch preview](https://github.com/dfe-analytical-services/analysts-guide/blob/main/CONTRIBUTING.md) is on the right in the screenshot below)

image

**Review by cjrace — CHANGES_REQUESTED** [review]

**Comment by cjrace** [comment]
Closing in favour of https://github.com/dfe-analytical-services/analysts-guide/pull/4

**Pull request closed without merging**